res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -44,76 +60,239 @@ DJCi300.scratchShiftMultiplier = 4;
 DJCi300.bendScale = 1.0;
 
 // Other scratch related options
-DJCi300.kScratchActionNone = 0;
+DJCi300.kScratchActionBend = 0;
 DJCi300.kScratchActionScratch = 1;
 DJCi300.kScratchActionSeek = 2;
-DJCi300.kScratchActionBend = 3;
+
+// Pad modes
+DJCi300.padModeNone = -1;
+DJCi300.padModeHotcue = 0;
+DJCi300.padModeRoll = 1;
+DJCi300.padModeSlicer = 2;
+DJCi300.padModeSampler = 3;
+DJCi300.padModeToneplay = 4;
+DJCi300.padModeFX = 5;
+DJCi300.padModeSlicerloop = 6;
+DJCi300.padModeBeatjump = 7;
+
+// Slicer connections
+let slicerSizeConnection1;
+let slicerTrackConnection1;
+let slicerBeatConnection1;
+let slicerTrackConnection2;
+let slicerSizeConnection2;
+let slicerBeatConnection2;
 
 DJCi300.vuMeterUpdateMaster = function(value, _group, _control) {
-    value = (value * 122) + 5;
+    value = (value * 125);
     midi.sendShortMsg(0xB0, 0x40, value);
     midi.sendShortMsg(0xB0, 0x41, value);
 };
 
 DJCi300.vuMeterUpdateDeck = function(value, group, _control, _status) {
-    value = (value * 122) + 5;
-    var status = (group === "[Channel1]") ? 0xB1 : 0xB2;
+    value = (value * 125);
+    const status = (group === "[Channel1]") ? 0xB1 : 0xB2;
     midi.sendShortMsg(status, 0x40, value);
 };
 
 DJCi300.init = function() {
     // Scratch button state
-    DJCi300.scratchButtonState = true;
+    DJCi300.scratchButtonState = {
+        1: true,
+        2: true
+    };
     // Scratch Action
     DJCi300.scratchAction = {
-    1: DJCi300.kScratchActionNone,
-    2: DJCi300.kScratchActionNone
+        1: DJCi300.kScratchActionBend,
+        2: DJCi300.kScratchActionBend
+    };
+    // Platter state (whether the jog wheel is pressed or not)
+    DJCi300.wheelTouchState = {
+        1: false,
+        2: false
+    };
+
+    // Pad mode variables
+    // Initialize to padModeNone
+    DJCi300.padMode = {
+        1: DJCi300.padModeNone,
+        2: DJCi300.padModeNone
+    };
+
+    // Toneplay variables
+    // Toneplay offset (shifts the toneplay keyboard up or down)
+    DJCi300.toneplayOffset = {
+        1: 0,
+        2: 0
+    };
+
+    // Slicer variables
+    // Slicer storage (stores slicer button positions)
+    DJCi300.slicerPoints = {
+        1: [-1, -1, -1, -1, -1, -1, -1, -1, -1],
+        2: [-1, -1, -1, -1, -1, -1, -1, -1, -1]
+    };
+    // Slicer buttons (stores whether or not slicer button is pressed)
+    DJCi300.slicerButtonEnabled = {
+        1: [0, 0, 0, 0, 0, 0, 0, 0],
+        2: [0, 0, 0, 0, 0, 0, 0, 0]
```

Comment:
these are booleans right? Then use those please:
```suggestion
        1: [false, false, false, false, false, false, false, false],
        2: [false, false, false, false, false, false, false, false]
```